internal/cloud/gke.go

```diff
@@ -660,6 +692,188 @@ func getAnnotation(p *corev1.Pod, key string) string {
 	return p.Annotations[key]
 }
 
+// StaticallyProvisionNodePools provisions node pools from a list of reservations.
+func (g *GKE) StaticallyProvisionNodePools(ctx context.Context, reservations []string) error {
+	log.Info("statically provisioning node pools for reservations", "reservations", reservations)
+	for _, reservationName := range reservations {
+		reservation, err := g.Reservations.GetReservation(ctx, reservationName)
+		if err != nil {
+			return fmt.Errorf("getting reservation %s: %w", reservationName, err)
+		}
+
+		if reservation == nil {
+			log.Info("reservation not found, skipping", "reservationName", reservationName)
+			continue
+		}
+
+		var chipCount int64
+		// Check for specific reservation properties
+		if reservation.SpecificReservation != nil &&
+			reservation.SpecificReservation.InstanceProperties != nil &&
+			len(reservation.SpecificReservation.InstanceProperties.GuestAccelerators) > 0 {
+			chipCount = reservation.SpecificReservation.InstanceProperties.GuestAccelerators[0].AcceleratorCount
+		} else if reservation.AggregateReservation != nil &&
+			len(reservation.AggregateReservation.ReservedResources) > 0 &&
+			reservation.AggregateReservation.ReservedResources[0].Accelerator != nil {
+			// This logic is based on the user's instruction that an AggregateReservation field exists.
+			chipCount = reservation.AggregateReservation.ReservedResources[0].Accelerator.AcceleratorCount
+		} else {
+			log.Info("reservation has no supported specific or aggregate reservation properties with accelerators, skipping", "reservationName", reservationName)
+			continue
+		}
+
+		if chipCount == 0 {
+			log.Info("could not determine chip count from reservation, skipping", "reservationName", reservationName)
+			continue
+		}
+
+		subBlockCount := chipCount / 64
+
+		for i := 0; i < int(subBlockCount); i++ {
+			nodePoolID := fmt.Sprintf("%s-%d", reservationName, i)
+			np, err := g.nodePoolForStaticReservation(nodePoolID, reservationName)
+			if err != nil {
+				return fmt.Errorf("determining node pool for static reservation: %w", err)
+			}
+			log.Info("determined node pool for static reservation", "nodePoolName", np.Name, "nodePool", np)
+
+			existingNPState, err := g.checkExistingNodePool(ctx, np)
```

Comment:
This func probably should not be used for this case. It is really for checking if a nodepool matches a given workload. We should just check for existance.